src/Boids.hs

```diff
@@ -149,10 +156,31 @@ initialState :: IO World
 initialState = do boids <- replicateM 200 randomBoid
                   return $ World { worldBoids = boids
                                  , worldGuide = (0.0, 0.0)
+                                 , worldViewPort = viewPortInit
+                                 , worldDraggingPivot = Nothing
                                  }
 
+zoom :: Float -> ViewPort -> ViewPort
+zoom step viewPort = viewPort {
```

Comment:
Actually move that outside of Boids module as well. Create a module for custom ViewPort operations. Other operations like translation should be implemented there too.